e3nn/o3/tensor_product.py

```diff
@@ -346,6 +348,11 @@ def indent_for_level(indent_level):
 
             if ins.has_weight:
                 index_w += 1
+                # Extract the weight from the flattened weight tensor
+                line = f"{s}ws_{index_w} = ws[..., {flat_weight_i}:{flat_weight_i + _prod(ins.weight_shape)}].reshape({'' if self.shared_weights else '-1, '}{', '.join(str(d) for d in ins.weight_shape)})\n"
```

Comment:
I think we can simply call it `ws` no need of `ws_{index_w}` ?

Reply:
it would have to be `ws_this` so it doesn't shadow the full weight tensor, but yeah, we could

maybe it's easier to understand with explicit indexes though? No need to figure out where the last line that redefined `ws_this` was. Not sure.

Reply:
indeed its cool also as it is now